contracts/Core/parameters/Governance.sol

```diff
@@ -0,0 +1,133 @@
+        stakeManagerParams.setWithdrawReleasePeriod(_withdrawReleasePeriod);
+    }
+
+    function setExtendLockPenalty(uint8 _extendLockPenalty) external initialized onlyRole(GOVERNER_ROLE) {
+        emit ParameterChanged(msg.sender, "extendLockPenalty", _extendLockPenalty, block.timestamp);
+        stakeManagerParams.setExtendLockPenalty(_extendLockPenalty);
+    }
+
+    function setMaxAltBlocks(uint8 _maxAltBlocks) external initialized onlyRole(GOVERNER_ROLE) {
+        emit ParameterChanged(msg.sender, "maxAltBlocks", _maxAltBlocks, block.timestamp);
+        blockManagerParams.setMaxAltBlocks(_maxAltBlocks);
+    }
+
+    function setMinStake(uint256 _minStake) external initialized onlyRole(GOVERNER_ROLE) {
+        emit ParameterChanged(msg.sender, "minStake", _minStake, block.timestamp);
+        stakeManagerParams.setMinStake(_minStake);
+        voteManagerParams.setMinStake(_minStake);
+        blockManagerParams.setMinStake(_minStake);
+    }
+
+    function setBlockReward(uint256 _blockReward) external initialized onlyRole(GOVERNER_ROLE) {
+        emit ParameterChanged(msg.sender, "blockReward", _blockReward, block.timestamp);
+        blockManagerParams.setBlockReward(_blockReward);
+        rewardManagerParams.setBlockReward(_blockReward);
+    }
+
+    function setGracePeriod(uint16 _gracePeriod) external initialized onlyRole(GOVERNER_ROLE) {
+        emit ParameterChanged(msg.sender, "gracePeriod", _gracePeriod, block.timestamp);
+        rewardManagerParams.setGracePeriod(_gracePeriod);
+        stakeManagerParams.setGracePeriod(_gracePeriod);
+    }
+
+    function setMaxAge(uint32 _maxAge) external initialized onlyRole(GOVERNER_ROLE) {
+        emit ParameterChanged(msg.sender, "maxAge", _maxAge, block.timestamp);
+        rewardManagerParams.setMaxAge(_maxAge);
+    }
+
+    function setMaxCommission(uint8 _maxCommission) external initialized onlyRole(GOVERNER_ROLE) {
+        require(_maxCommission <= 100, "Invalid Max Commission Update");
+        emit ParameterChanged(msg.sender, "maxCommission", _maxCommission, block.timestamp);
+        stakeManagerParams.setMaxCommission(_maxCommission);
+    }
+
+    function disableEscapeHatch() external initialized onlyRole(GOVERNER_ROLE) {
+        emit ParameterChanged(msg.sender, "escapeHatchEnabled", 0, block.timestamp);
+        stakeManagerParams.disableEscapeHatch();
+    }
+
+    function setEpochLength(uint16 _epochLength) external initialized onlyRole(GOVERNER_ROLE) {
+        emit ParameterChanged(msg.sender, "epochLength", _epochLength, block.timestamp);
+        blockManagerParams.setEpochLength(_epochLength);
+        rewardManagerParams.setEpochLength(_epochLength);
+        stakeManagerParams.setEpochLength(_epochLength);
+        voteManagerParams.setEpochLength(_epochLength);
+        assetManagerParams.setEpochLength(_epochLength);
+        delegatorParams.setEpochLength(_epochLength);
+        randomNoManagerParams.setEpochLength(_epochLength);
```

Comment:
I agree, that this is not neat, 
as epochLength is being used by StateManager only
so one can say we should have it as param there only.

issue is with dimond inheritance.

if we inherit same GovernanceACL.sol in stateManager then in any main contract, two copies of them get inherited 
so we have to go this way.
one from stateManager and one from their own param child
let me know if there is a way